cpp/examples/multi_gpu/mg_examples.cpp

```diff
@@ -0,0 +1,180 @@
+  auto const comm_size = comm.get_size();
+
+  auto gpu_row_comm_size = static_cast<int>(sqrt(static_cast<double>(comm_size)));
+  while (comm_size % gpu_row_comm_size != 0) {
+    --gpu_row_comm_size;
+  }
+
+  cugraph::partition_manager::init_subcomm(*handle, gpu_row_comm_size);
+
+  return std::move(handle);
+}
+
+/**
+ * @brief This function reads graph from an input csv file and run BFS and Louvain on it.
+ */
+
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+void run_graph_algos(raft::handle_t const& handle, std::string const& csv_graph_file_path)
+{
+  auto const comm_rank = handle.get_comms().get_rank();
+  auto const comm_size = handle.get_comms().get_size();
+
+  std::cout << "Rank_" << comm_rank << ", reading graph from " << csv_graph_file_path << std::endl;
+  bool renumber = true;  // must be true for distributed graph.
+  auto [graph, edge_weights, renumber_map] =
+    cugraph::test::read_graph_from_csv_file<vertex_t, edge_t, weight_t, false, multi_gpu>(
+      handle, csv_graph_file_path, true, renumber);
+
+  auto graph_view       = graph.view();
+  auto edge_weight_view = edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+  assert(graph_view.local_vertex_partition_range_size() == (*renumber_map).size());
+
+  // run example algorithms
+
+  // BFS
+
+  rmm::device_uvector<vertex_t> d_distances(graph_view.local_vertex_partition_range_size(),
+                                            handle.get_stream());
+  rmm::device_uvector<vertex_t> d_predecessors(graph_view.local_vertex_partition_range_size(),
+                                               handle.get_stream());
+
+  rmm::device_uvector<vertex_t> d_sources(1, handle.get_stream());
+  std::vector<vertex_t> h_sources = {0};
+  raft::update_device(d_sources.data(), h_sources.data(), h_sources.size(), handle.get_stream());
+
+  cugraph::bfs(handle,
+               graph_view,
+               d_distances.data(),
+               d_predecessors.data(),
+               d_sources.data(),
+               d_sources.size(),
+               false,
+               std::numeric_limits<vertex_t>::max());
+
+  // Louvain
+
+  rmm::device_uvector<vertex_t> cluster_assignments(graph_view.local_vertex_partition_range_size(),
+                                                    handle.get_stream());
+
+  weight_t threshold  = 1e-7;
+  weight_t resolution = 1.0;
+  size_t max_level    = 10;
+
+  weight_t modularity{-1.0};
+  std::tie(std::ignore, modularity) =
+    cugraph::louvain(handle,
+                     std::optional<std::reference_wrapper<raft::random::RngState>>{std::nullopt},
+                     graph_view,
+                     edge_weight_view,
+                     cluster_assignments.data(),
+                     max_level,
+                     threshold,
+                     resolution);
+
+  std::cout << "rank : " << comm_rank << ", modularity : " << modularity << std::endl;
```

Comment:
Maybe also display the cluster assignments with `raft::print_device_vector`?

Reply:
Perhaps add `std::min(cluster_assignments.size(), size_t{20})` as the size in the `print_device_vector` call to demonstrate that you can call it on a subset of the device vector... in case they run a larger graph through the test code.